### PR TITLE
Do not allow to set higher permissions on a public link share for a r…

### PR DESCRIPTION
**Pull request by DeepDiver1975**
…esource which was shared with limited permissions

## Description
As the acceptance scenario shows a public link is not allowed to have more permissions then the original share.

## Related Issue
- refs owncloud/enterprise#3364

## How Has This Been Tested?
- running acceptance tests

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: apps/files_sharing/lib/Controller/Share20OcsController.php [hunk 1]
```diff
@@ -825,6 +825,14 @@ public function updateShare($id) {
 			}
 
 			if ($newPermissions !== null) {
+				$shareHome = $this->rootFolder->getUserFolder($share->getSharedBy());
+				$nodes = $shareHome->getById($share->getNode()->getId());
+				foreach ($nodes as $node) {
+					if (($node->getPermissions() | $newPermissions) !== $node->getPermissions()) {
+						return new Result(null, 404, 'Cannot increase permission of ' . $share->getTarget());
```
Comment (IljaN) [inline apps/files_sharing/lib/Controller/Share20OcsController.php]: Can we use 403 instead 404 here?

Comment (phil-davis) [reply apps/files_sharing/lib/Controller/Share20OcsController.php]: Lots of the scenarios in `reShare.feature` test sharing requests that are not allowed. The code currently returns 404 for all those. I have no idea why. IMO "403 - forbidden" seems the more appropriate code. But if it is changed here, then it should be changed for all the other permissions check cases, and that is a bigger change which could break clients that have specifically coded to know about 404.

Comment (DeepDiver1975) [reply apps/files_sharing/lib/Controller/Share20OcsController.php]: 403 looks more reasonable - yes

In order to stay consistent with all other error cases of the same kind I prefer to stick with 404

Comment (patrickjahns) [reply apps/files_sharing/lib/Controller/Share20OcsController.php]: Agreed - let's stay consistent with 404 for now.
We can create a separate PR with api changes for upcoming releases that move theses error scenarios into a 403 if desired

**Review by IljaN — DISMISSED** [review]

**Review by phil-davis — COMMENTED** [review]
Adjusted acceptance test is good, it passes and all other CI passes. So it must be good ;)
Someone else can review-approve the technical implementation code.

**Review by IljaN — APPROVED** [review]

**Comment by phil-davis** [comment]
Backports:
`stable10` #35599 
`release-10.2.1` #35600

**Pull request merged**